### PR TITLE
Feature/child info page

### PR DESCRIPTION
**Pull request by liding987**
### What was the feature/problem?

{Please write here}

### How does this PR address the above feature/problem?

{Please write here}

### Check lists (check `x` in `[ ]` of list items)

- [ ] Test passed
- [ ] Coding style (indentation, etc)

### Additional Comments (if any)

{Please write here}

### CODE REVIEW

File: ui/src/childinfo/ChildInfo.js [hunk 1]
```diff
@@ -1,177 +1,204 @@
...
+          name: this.state.name,
+          age: this.state.age,
+          hometown: this.state.homeTown,
+          illness: this.state.illness
         },
-        3: {
-          text: 'Tell us more about your wish!',
-          input: 'details'
-        }
+        type: "go",
```
Comment (jrgoodpasture) [inline ui/src/childinfo/ChildInfo.js]: This should be gotten from props.type, not hardcoded

File: ui/src/childinfo/ChildInfo.js [hunk 2]
```diff
@@ -1,177 +1,204 @@
...
+      this.setState({
+        step: step + 1
+      }, () => {
+        this.scrollToTop();
+      });
+    } else {
+
+      const wish = {
+        child: {
+          name: this.state.name,
```
Comment (jrgoodpasture) [inline ui/src/childinfo/ChildInfo.js]: We should destructure all these variables to be consistent with rest of the project

File: ui/src/childinfo/ChildInfo.js [hunk 3]
```diff
@@ -1,177 +1,204 @@
...
         },
-        3: {
-          text: 'Tell us more about your wish!',
-          input: 'details'
-        }
+        type: "go",
+        details: this.state.details
+      }
+
+      WishDetailsService.makeAWish(wish)
```
Comment (jrgoodpasture) [inline ui/src/childinfo/ChildInfo.js]: Think we will need to be doing something with the _id value from this response

File: ui/src/childinfo/ChildInfo.test.js [hunk 4]
```diff
@@ -8,8 +8,9 @@ describe('Initial Render', () => {
   const testName = 'John'
   const age = 5;
   let childInfo;
+  let childId = 'some id string'
   beforeEach(() => {
-    childInfo = shallow(<ChildInfo name={testName} age={age}/>);
+    childInfo = shallow(<ChildInfo name={testName} age={age} childId={childId} />);
```
Comment (jrgoodpasture) [inline ui/src/childinfo/ChildInfo.test.js]: Don't think we will have childId at this point, that will be received after a post to makeAWish

File: ui/src/childinfo/ChildInfo.test.js [hunk 5]
```diff
@@ -65,22 +66,36 @@ describe('Initial Render', () => {
...
+    it('Should show WishDetails component', () => {
+      expect(childInfo.find('WishDetails').length).toEqual(1)
+    })
+  })
+
   describe('When user on confirmation page and click blast off the rocket', () => {
 
     beforeEach(() => {
-      childInfo = shallow(<ChildInfo name={testName} age={age}/>);
+      childInfo = shallow(<ChildInfo name={testName} age={age} childId={childId} />);
```
Comment (jrgoodpasture) [inline ui/src/childinfo/ChildInfo.test.js]: Again dont think the childId is available here/ shouldnt be in props

File: ui/src/childinfo/ChildInfo.js [hunk 6]
```diff
@@ -1,180 +1,214 @@
...
-          text: 'Tell us more about your wish!',
-          input: 'details'
-        }
+        type: type ? type : '',
+        details: details
+      }
+
+      const response = await WishDetailsService.makeAWish(wish)
+
+      debugger
```
Comment (jrgoodpasture) [inline ui/src/childinfo/ChildInfo.js]: Should remove debugger point

File: ui/src/childinfo/ChildInfo.test.js (hunk 4 above)
Comment (jrgoodpasture) [reply ui/src/childinfo/ChildInfo.test.js]: This still needs to be removed

File: ui/src/childinfo/ChildInfo.test.js [hunk 7]
```diff
@@ -49,40 +79,63 @@ describe('Initial Render', () => {
...
-      nextButton.simulate('click');
+      await nextButton.simulate('click');
       expect(childInfo.find('.text-name').text()).toEqual('Tell us about your condition');
-      nextButton.simulate('click');
+      await nextButton.simulate('click');
       expect(childInfo.find('.text-name').text()).toEqual('Tell us more about your wish!');
-      nextButton.simulate('click');
+      await nextButton.simulate('click');
+
+      console.log(childInfo.find('.text-name').debug())
```
Comment (jrgoodpasture) [inline ui/src/childinfo/ChildInfo.test.js]: Should remove this log

File: ui/src/childinfo/ChildInfo.test.js [hunk 8]
```diff
@@ -49,40 +79,63 @@ describe('Initial Render', () => {
...
-      childInfo = shallow(<ChildInfo name={testName} age={age}/>);
+      childInfo = shallow(<ChildInfo name={testName} age={age} />);
     })
 
-    it('Should update text on button click', () => {
+    it('Should update text on button click', async () => {
       let nextButton = childInfo.find('.next-button');
       expect(childInfo.find('.text-name').text()).toEqual(`Hi ${testName}, I have a few questions for you before we can make your wish come true!`);
-      nextButton.simulate('click');
+      await nextButton.simulate('click');
```
Comment (jrgoodpasture) [inline ui/src/childinfo/ChildInfo.test.js]: Are awaits really necessary here?

Comment (liding987) [reply ui/src/childinfo/ChildInfo.test.js]: because the next button has an async call

**Pull request merged**